From 98ae35cab0e40e59963f9f58345bf378b9522d74 Mon Sep 17 00:00:00 2001
From: Andrew Cooper <andrew.cooper3@citrix.com>
Date: Mon, 22 Jan 2024 14:50:10 +0000
Subject: x86/entry: Fix ELF metadata for NMI and handle_ist_exception

handle_ist_exception isn't part of the NMI handler, just like handle_exception
isn't part of #PF.

Fixes: b3a9037550df ("x86: annotate entry points with type and size")
Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
Reviewed-by: Jan Beulich <jbeulich@suse.com>

diff --git a/xen/arch/x86/x86_64/entry.S b/xen/arch/x86/x86_64/entry.S
index cd4a9e4ee855..56811a3d68ac 100644
--- a/xen/arch/x86/x86_64/entry.S
+++ b/xen/arch/x86/x86_64/entry.S
@@ -1044,7 +1044,9 @@ FUNC(nmi)
         ENDBR64
         pushq $0
         movl  $TRAP_nmi, EFRAME_entry_vector(%rsp)
-handle_ist_exception:
+END(nmi)
+
+FUNC(handle_ist_exception)
         ALTERNATIVE "", clac, X86_FEATURE_XEN_SMAP
         SAVE_ALL
 
@@ -1172,7 +1174,7 @@ handle_ist_exception:
         ASSERT_CONTEXT_IS_XEN
         jmp   restore_all_xen
 #endif
-END(nmi)
+END(handle_ist_exception)
 
 FUNC(machine_check)
         ENDBR64
